### PR TITLE
Feature/suic 84/dropdown option

### PR DESCRIPTION
**Pull request by danivm**
Create new `DropdownOption` component.

👀 [Demo](https://sui-components-pr.now.sh/workbench/molecule/dropdownOption/demo)
🗒   [Jira](https://jira.scmspain.com/browse/SUIC-84)
🎨   [Paper](https://paper.dropbox.com/doc/SUI-Dropdown-list--ARKBccUZsqdxDlEy7T5kR44DAg-VSQSdXvEyqawESsk4IkMC)

### CODE REVIEW

File: components/molecule/dropdownOption/src/_settings.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+$c-dropdown-option-disabled: #dcdcdc !default;
```
Comment (jcger) [inline components/molecule/dropdownOption/src/_settings.scss]: this is a lighten gray_l3, please check the color-variation function

I'm not sure if it should be `color-variation($c-gray, 3)`

File: components/molecule/dropdownOption/src/index.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+const CLASS = 'sui-MoleculeDropdownOption'
+
+class MoleculeDropdownOption extends Component {
+  state = {
+    selected: this.props.selected
+  }
+
+  handleSelect = () => {
+    if (!this.props.disabled) {
+      this.setState(
```
Comment (davecarter) [inline components/molecule/dropdownOption/src/index.js:15]: 👏 👏 👏

**Review by davecarter — DISMISSED** [review]

File: components/molecule/dropdownOption/src/index.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        prevState => ({
+          selected: !prevState.selected
+        }),
+        this.props.onChange()
```
Comment (carlosvillu) [inline components/molecule/dropdownOption/src/index.js]: Without params ?!

**Review by carlosvillu — DISMISSED** [review]

**Review by carlosvillu — DISMISSED** [review]

**Review by jcger — CHANGES_REQUESTED** [review]

File: components/molecule/dropdownOption/src/index.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  disabled: PropTypes.bool,
+
+  /** onClick callback */
+  onClick: PropTypes.func,
+
+  /** Is selected */
+  selected: PropTypes.bool,
+
+  /** Text */
+  text: PropTypes.string.isRequired
```
Comment (jcger) [inline components/molecule/dropdownOption/src/index.js]: please reconsider passing text by as children in order to be more compliant with html standards

We should follow this structure

```
<select>
  <option>1</option>
  <option selected>2</option>
</select>
```

**Review by jcger — DISMISSED** [review]

**Review by ddmarin94 — DISMISSED** [review]

File: components/molecule/dropdownOption/src/index.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    }
+  }
+
+  render() {
+    const {selected} = this.state
+    const {children, checkbox, disabled} = this.props
+    const wrapperClassName = cx(CLASS, {
+      [`${CLASS}-checkbox`]: checkbox,
+      [`${CLASS}--disabled`]: disabled,
+      [`${CLASS}--selected`]: selected
```
Comment (midudev) [inline components/molecule/dropdownOption/src/index.js]: Selected is not a modifier, is a state. For SUIT, it should follow `is-selected`.

File: components/molecule/dropdownOption/src/index.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
+import React, {Component} from 'react'
+import PropTypes from 'prop-types'
+import cx from 'classnames'
+import AtomInput from '@s-ui/react-atom-input'
+
+const CLASS = 'sui-MoleculeDropdownOption'
+
+class MoleculeDropdownOption extends Component {
+  state = {
+    selected: this.props.selected
```
Comment (midudev) [inline components/molecule/dropdownOption/src/index.js]: I would change the name of the prop from `selected` to `initialSelected` so it clearly states that the value of the initialSelected is what is going to be used to initialize the state but future updates are not going to update the component.

**Comment by midudev** [comment]
I've added some comments but overall it looks good. Fantastic work @danivm 👏

File: components/molecule/dropdownOption/src/index.js (hunk 6 above)
Comment (devictoribero) [reply components/molecule/dropdownOption/src/index.js]: I agree with @miduga . If not, what about `defaultSelected` or `defaultOption`?

File: components/molecule/dropdownOption/src/index.js [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+    }
+  }
+
+  render() {
+    const {selected} = this.state
+    const {children, checkbox, disabled} = this.props
+    const wrapperClassName = cx(CLASS, {
+      [`${CLASS}-checkbox`]: checkbox,
+      [`${CLASS}--disabled`]: disabled,
+      [`is-selected`]: selected
```
Comment (midudev) [inline components/molecule/dropdownOption/src/index.js]: Not important, but this could be simplified to `'is-selected': selected`

**Review by midudev — DISMISSED** [review]

**Review by ddmarin94 — DISMISSED** [review]

**Review by devictoribero — DISMISSED** [review]

File: demo/molecule/dropdownOption/demo/index.js [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+  render() {
+    return (
+      <div className="DemoMoleculeDropdownOption">
+        <h1>MoleculeDropdownOption</h1>
+        <div className="DemoMoleculeDropdownOption-section">
+          <h2>As a default option</h2>
+          <div className="DemoMoleculeDropdownOption-option">
+            <MoleculeDropdownOption
+              text="Option 1"
```
Comment (jcger) [inline demo/molecule/dropdownOption/demo/index.js]: You should remove this prop in all examples as it isn't used anymore

File: components/molecule/dropdownOption/README.md [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+$ npm install @s-ui/react-molecule-dropdown-option --save
+```
+
+## Usage
+
+### Basic usage
+```js
+import MoleculeDropdownOption from '@s-ui/react-molecule-dropdown-option'
+
+return (<MoleculeDropdownOption text="Option 1"/>)
```
Comment (jcger) [inline components/molecule/dropdownOption/README.md]: please update, text is not used anymore

**Review by jcger — APPROVED** [review]

**Review by midudev — APPROVED** [review]

**Pull request merged**